### ISSUE TITLE
15.8.1 Angular jest test started failing

### ISSUE THREAD

**Issue by Olster** [nrwl/nx#15390]
### Current Behavior

An Angular jest test that was working in Nx 15.7.2 started failing in Nx 15.8.1.
This same test runs successfully using karma.
If this is not a bug in Nx, please direct me to the right package that might be causing this. 

### Repo generated with
```plaintext
✔ Choose what to create                 · integrated
✔ What to create in the new workspace   · angular-monorepo
✔ Repository name                       · nx-test
✔ Application name                      · jest-bug
✔ Default stylesheet format             · scss
✔ Enable distributed caching to make your CI faster · No
```

### Command run
`yarn nx test jest-bug`

### Failing test
Added into `apps/jest-bug/src/app/app.component.spec.ts`

```typescript
async function testMe(): Promise<void> {
  try {
    await firstValueFrom(
      of('foo')
    );
  } catch (e) {
    console.error(e);
  } finally {
    console.log('Finally!');
  }
}

fdescribe('my test', () => {
  it('should work', fakeAsync(() => {
    const spy = jest.fn();

    console.log('Call testMe');
    testMe().then(spy);

    console.log('Tick');
    tick();

    console.log('Expect');
    expect(spy).toHaveBeenCalled();
  }));
});
```

### Behaviour

Test fails with a report
```plaintext
expect(jest.fn()).toHaveBeenCalled()

    Expected number of calls: >= 1
    Received number of calls:    0

      58 |
      59 |     console.log('Expect');
    > 60 |     expect(spy).toHaveBeenCalled();
         |                 ^
      61 |   }));
      62 | });
```

Console log seems to be out of order too:
```plaintext
console.log
    Call testMe

      at src/app/app.component.spec.ts:53:13

  console.log
    Tick

      at src/app/app.component.spec.ts:56:13

  console.log
    Expect

      at src/app/app.component.spec.ts:59:13

  console.log
    Finally!

      at testMe (src/app/app.component.spec.ts:45:13)
```

### Expected Behavior

In Nx 15.7.2 test runs successfully.

### Logs in Nx 15.7.2
```plaintext
console.log
    Call testMe

      at src/app/app.component.spec.ts:52:13

  console.log
    Tick

      at src/app/app.component.spec.ts:55:13

  console.log
    Finally!

      at src/app/app.component.spec.ts:44:13

  console.log
    Expect

      at src/app/app.component.spec.ts:58:13
```

### Steps to Reproduce

1. Run `npx create-nx-workspace@15.8.1 --package-manager yarn` with following settings:
```plaintext
✔ Choose what to create                 · integrated
✔ What to create in the new workspace   · angular-monorepo
✔ Repository name                       · nx-test
✔ Application name                      · jest-bug
✔ Default stylesheet format             · scss
✔ Enable distributed caching to make your CI faster · No
```
2. Add following test to `apps/jest-bug/src/app/app.component.spec.ts`
```typescript
async function testMe(): Promise<void> {
  try {
    await firstValueFrom(
      of('foo')
    );
  } catch (e) {
    console.error(e);
  } finally {
    console.log('Finally!');
  }
}

fdescribe('my test', () => {
  it('should work', fakeAsync(() => {
    const spy = jest.fn();

    console.log('Call testMe');
    testMe().then(spy);

    console.log('Tick');
    tick();

    console.log('Expect');
    expect(spy).toHaveBeenCalled();
  }));
});
```
3. Run `yarn nx test jest-bug`

### Nx Report

```shell
>  NX   Report complete - copy this into the issue template

   Node : 18.12.0
   OS   : darwin arm64
   yarn : 1.22.19
   
   nx                      : 15.8.1
   @nrwl/js                : 15.8.1
   @nrwl/jest              : 15.8.1
   @nrwl/linter            : 15.8.1
   @nrwl/workspace         : 15.8.1
   @nrwl/angular           : 15.8.1
   @nrwl/cli               : 15.8.1
   @nrwl/cypress           : 15.8.1
   @nrwl/devkit            : 15.8.1
   @nrwl/eslint-plugin-nx  : 15.8.1
   @nrwl/tao               : 15.8.1
   @nrwl/webpack           : 15.8.1
   typescript              : 4.9.5
```

**Comment by Olster** [comment]
I was able to simplify the function where test fails, it doesn't involve RxJS:
```typescript
async function testMe(): Promise<void> {
  try {
    console.log(await Promise.resolve('foo'));
  } catch (e) {
    console.error(e);
  } finally {
    console.log('Finally!');
  }
}
```

The test code is the same and also fails.

**Comment by barbados-clemens** [comment]
good news is I can reproduce it, bad news I have zero reason why.

some notes:
this happens even if Nx isn't involved. i.e. just running the jest CLI. so it most likely isn't an nx specific issue. 

My assumption right now is a regression in [jest-preset-angular](https://github.com/thymikee/jest-preset-angular) since it includes the Zone.js setup for jest. 
https://github.com/thymikee/jest-preset-angular/blob/main/setup-jest.js


Maybe partially related: https://github.com/thymikee/jest-preset-angular/issues/1977

**Comment by Olster** [comment]
Thanks for checking it out @barbados-clemens !
Can confirm, test runs successfully in `jest-preset-angular` v12.2.6 but fails in v13.0.0.
Opened an issue there: https://github.com/thymikee/jest-preset-angular/issues/1994

I'm going to take a look if I can find the cause of this issue. Not sure if this issue for nx should be closed, I assume quite a few people can run into this issue after updating.

**Comment by k3nsei** [comment]
For me downleveling tests to `es2015` in `tsconfig.spec.json` fixed problem with tests. It had something to do with fact that `zone.js` can't patch `async/await` in my case. But if target is set to `es2015` typescript is downleveling `async/await` to iterators/generators. Maybe it's same problem here.

**Comment by Olster** [comment]
@k3nsei  the test inside `jest-preset-angular` package fails even with `"module": "CommonJS",`
In the repo where original issue occurred we're also using commonjs

I also see that Angular started adding zone.js to karma tests differently in September of 2022:
https://github.com/angular/angular-cli/commit/56a1e8f9f52658488afb9d36007e96c96d08a03b
They're importing `zone.js` and `zone.js/testing` as polyfills into karma tests

**Comment by barbados-clemens** [comment]
can confirm using es2015 in the tsconfig.spec.json didn't fix the issue in the reproducible.
Messing around with various zone.js imports didn't solve anything for me, but I'm no angular expert so kind of just trying a bunch of things, so might not be exactly a jest-preset-angular issue, but they might be able to point us in a better direction.

**Comment by Olster** [comment]
@k3nsei was right, I changed the module to es2015 instead of changing the target.
`jest-preset-angular` was downgrading the target automatically and doesn't do that anymore: https://github.com/thymikee/jest-preset-angular/commit/695c73022c4011cca477b8d6d62c174a8c01a9ec

Credit to @dzonatan for figuring this out in the jest-preset-angular bug report.

Changing the target to es2015 or even es2016 fixed the problem.
Closing the issue since it's not related to nx.
